clash-vexriscv-sim/data/Example2.scala

```diff
@@ -0,0 +1,135 @@
+// SPDX-FileCopyrightText: 2022-2024 Google LLC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package cpu
+
+import spinal.core._
+import spinal.lib._
+import spinal.lib.com.jtag.Jtag
+import spinal.lib.cpu.riscv.debug._
+import vexriscv.plugin._
+import vexriscv.{VexRiscv, VexRiscvConfig, plugin}
+import vexriscv.ip.{DataCacheConfig}
+import vexriscv.ip.fpu.FpuParameter
+
+object Example2 extends App {
+  def cpu() : VexRiscv = {
+    val config = VexRiscvConfig(
+      plugins = List(
+        new IBusSimplePlugin(
+          resetVector = 0x20000000l,
+          cmdForkOnSecondStage = false,
+          cmdForkPersistence = false,
+          prediction = NONE,
+          catchAccessFault = true,
+        ),
+
+        new DBusCachedPlugin(
+          /*
+          config = new DataCacheConfig(
+            cacheSize        = 2048,
+            bytePerLine      = 32,
+            wayCount         = 1,
+            addressWidth     = 32,
+            cpuDataWidth     = 32,
+            memDataWidth     = 32,
+            catchAccessError = true,
+            catchIllegal     = true,
+            catchUnaligned   = true
+          )
+          */
+          config = new DataCacheConfig(
+            cacheSize        = 8,
+            bytePerLine      = 8,
+            wayCount         = 1,
+            addressWidth     = 32,
+            cpuDataWidth     = 32,
+            memDataWidth     = 32,
+            catchAccessError = true,
+            catchIllegal     = true,
+            catchUnaligned   = true
+          )
```

Comment:
Can this cause type mismatches between the generated verilog and haskell types? e.g different datawidths or addresswidths?